gtars/src/scoring/files.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use glob::glob;
+use rust_lapper::{Interval, Lapper};
+
+use crate::common::models::region_set::RegionSet;
+use crate::common::models::Region;
+use crate::common::utils::generate_region_to_id_map;
+
+#[allow(unused)]
+pub struct OverlapResult(Region, pub(crate) u32);
+
+pub trait FindOverlaps {
+    fn find_overlaps(&self, region: &Region) -> Option<Vec<OverlapResult>>;
+}
+
+pub struct FragmentFileGlob {
+    curr: usize,
+    files: Vec<PathBuf>,
+}
+
+pub struct ConsensusSet {
+    len: usize,
+    overlap_trees: HashMap<String, Lapper<u32, u32>>,
+}
+
+impl FragmentFileGlob {
+    pub fn new(pattern: &str) -> Result<Self> {
+        let files = glob(pattern)?;
+        let files = files
+            .map(|f| match f {
+                Ok(path) => Ok(path),
+                Err(_) => anyhow::bail!(format!("Error reading file entry: {:?}", f)),
+            })
+            .collect::<Result<Vec<_>>>()?;
+        let curr = 0_usize;
```

Comment:
```suggestion
        let curr = 0;
```
compiler should infer this